step-cli/step-cli-core/src/main/java/step/cli/AbstractCliTool.java

```diff
@@ -20,23 +20,26 @@
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import step.automation.packages.client.RemoteAutomationPackageClientImpl;
 import step.automation.packages.client.model.AutomationPackageSource;
-import step.automation.packages.client.model.FileSource;
-import step.automation.packages.client.model.MavenSnippetSource;
+import step.cli.parameters.Parameters;
 import step.client.AbstractRemoteClient;
 import step.client.credentials.ControllerCredentials;
 import step.core.maven.MavenArtifactIdentifier;
 
 import java.io.File;
 
-public abstract class AbstractCliTool implements CliToolLogging {
+public abstract class AbstractCliTool<T extends Parameters> implements CliToolLogging {
 
     private static final Logger log = LoggerFactory.getLogger(AbstractCliTool.class);
 
     private String url;
 
-    public AbstractCliTool(String url) {
+    protected final T parameters;
+
+    public AbstractCliTool(String url, T parameters) {
         this.url = url;
+        this.parameters = parameters;
```

Comment:
Just to be sure -- `this.parameters = Objects.requireNonNull(parameters)`, I assume.